src/animal/animal.service.ts

```diff
@@ -78,10 +76,14 @@ export class AnimalService implements IService<AnimalModel, AnimalModel> {
    * @param data
    * @returns
    */
-  async create(data: AnimalModel): Promise<AnimalModel> {
-    if (await this.animalServiceInfo.getAnimalInfo(data.type)) {
+  async create(data: Animal): Promise<Animal> {
+    const existByName = await this.getAnimalInfo(data.type);
+    if (existByName) {
       return this.repository.create(data);
-    } else throw new NotFoundException('Animal type not exist, please insert a new one');
+    } else {
+      this.logger.error("Cannot create or update the animal inserted 'cause it doesn't exist");
+      throw new UnprocessableEntity();
```

Comment:
You are in the service that is part of the domain, you shouldn't use types that depend on external adapters or transport layers. Instead, use a dedicated 'Error' exception.